robottelo/api/utils.py

```diff
@@ -74,6 +76,92 @@ def upload_manifest(organization_id, manifest):
+    :param str repo_url: Url of the repo that can be synced using pulp:
+        pulp repo or puppet forge.
+    :param organization_id: Organization id that is shared between
+        created entities.
+    :return: `nailgun.entities.ContentView` entity.
+    """
+    if not organization_id:
+        organization_id = entities.Organization().create().id
+    # Create product and puppet modules repository
+    product = entities.Product(organization=organization_id).create()
+    repo = entities.Repository(
+        product=product,
+        content_type='puppet',
+        url=repo_url
+    ).create()
+    # Synchronize repo via provided URL
+    repo.sync()
+    # Add selected module to Content View
+    cv = entities.ContentView(organization=organization_id).create()
+    for module in puppet_modules:
+        entities.ContentViewPuppetModule(
+            author=module['author'],
+            name=module['name'],
+            content_view=cv,
+        ).create()
+    # CV publishing will automatically create Environment and
+    # Puppet Class entities
+    cv.publish()
+    return cv.read()
+
+
+def delete_puppet_class(puppetclass_name, puppet_module=None,
+                        proxy_hostname=None, environment_name=None):
+    """Removes puppet class entity and uninstall puppet module from Capsule if
+    puppet module name and Capsule details provided.
+
+    :param str puppetclass_name: Name of the puppet class entity that should be
+        removed.
+    :param str puppet_module: Name of the module that should be
+        uninstalled via puppet.
+    :param str proxy_hostname: Hostname of the Capsule from which puppet module
+        should be removed.
+    :param str environment_name: Name of environment where puppet module was
+        imported.
+    """
+    # Find puppet class
+    puppet_classes = entities.PuppetClass().search(
+        query={'search': 'name = "{0}"'.format(puppetclass_name)}
+    )
+    # And all subclasses
+    puppet_classes.extend(entities.PuppetClass().search(
+        query={'search': 'name ~ "{0}::"'.format(puppetclass_name)})
+    )
+    for puppet_class in puppet_classes:
+        # Search and remove puppet class from affected hostgroups
+        for hostgroup in puppet_class.read().hostgroup:
+            hostgroup.delete_puppetclass(
+                data={'puppetclass_id': puppet_class.id}
+            )
+        # Search and remove puppet class from affected hosts
+        for host in entities.Host().search(
+                query={'search': 'class={0}'.format(puppet_class.name)}):
+            host.delete_puppetclass(
+                data={'puppetclass_id': puppet_class.id}
+            )
+        # Remove puppet class entity
+        puppet_class.delete()
+    # And remove puppet module from the system if puppet_module name provided
+    if puppet_module and proxy_hostname and environment_name:
+        ssh.command(
+            'puppet module uninstall --force {0}'.format(puppet_module))
```

Comment:
`ssh.command` only captures the output but doesn't verify it. In case command fails it'll be hard to find the root cause of failure as we're not verifying command result here. Mind adding either assertion or raising exception here?